test/ref_ops_test.cpp

```diff
@@ -4612,6 +4612,72 @@ TEST_CASE(neg_dyn_test)
     EXPECT(migraphx::verify_range(result_vector, gold));
 }
 
+TEST_CASE(nms_sanity_compute_test)
+{
+    migraphx::program p = migraphx::parse_onnx("nms_test.onnx");
+    p.compile(migraphx::ref::target{});
+
+    migraphx::shape mob_shape{migraphx::shape::int64_type, {1}};
+    std::vector<uint64_t> mob_data = {0};
+
+    migraphx::shape boxes_shape{migraphx::shape::float_type, {1, 6, 4}};
+    std::vector<float> boxes_data = {1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+                                     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1};
+
+    migraphx::shape score_shape{migraphx::shape::float_type, {1, 1, 6}};
+    std::vector<float> score_data = {1, 2, 3, 4, 5, 6};
+
+    migraphx::shape score_thresh_shape{migraphx::shape::float_type, {1}};
+    std::vector<float> score_thresh = {0.0f};
+
+    migraphx::shape iou_thresh_shape{migraphx::shape::float_type, {1}};
+    std::vector<float> iou_thresh = {0.0f};
+
+    migraphx::parameter_map pp;
+    pp["max_output_boxes_per_class"] = migraphx::argument(mob_shape, mob_data.data());
+    pp["boxes"]                      = migraphx::argument(boxes_shape, boxes_data.data());
+    pp["scores"]                     = migraphx::argument(score_shape, score_data.data());
+    pp["score_threshold"]            = migraphx::argument(score_thresh_shape, score_thresh.data());
+    pp["iou_threshold"]              = migraphx::argument(iou_thresh_shape, iou_thresh.data());
+
+    p.eval(pp).back();
+
+    EXPECT(true);
+}
+
+TEST_CASE(nms_large_input_boxes_test)
+{
+    migraphx::program p = migraphx::parse_onnx("nms_test.onnx");
+    p.compile(migraphx::ref::target{});
+
+    migraphx::shape mob_shape{migraphx::shape::int64_type, {1}};
+    std::vector<uint64_t> mob_data = {9223372036854775807};
+
+    migraphx::shape boxes_shape{migraphx::shape::float_type, {1, 6, 4}};
+    std::vector<float> boxes_data = {1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+                                     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1};
+
+    migraphx::shape score_shape{migraphx::shape::float_type, {1, 1, 6}};
+    std::vector<float> score_data = {1, 2, 3, 4, 5, 6};
+
+    migraphx::shape score_thresh_shape{migraphx::shape::float_type, {1}};
+    std::vector<float> score_thresh = {0.0f};
+
+    migraphx::shape iou_thresh_shape{migraphx::shape::float_type, {1}};
+    std::vector<float> iou_thresh = {0.0f};
+
+    migraphx::parameter_map pp;
+    pp["max_output_boxes_per_class"] = migraphx::argument(mob_shape, mob_data.data());
+    pp["boxes"]                      = migraphx::argument(boxes_shape, boxes_data.data());
+    pp["scores"]                     = migraphx::argument(score_shape, score_data.data());
+    pp["score_threshold"]            = migraphx::argument(score_thresh_shape, score_thresh.data());
+    pp["iou_threshold"]              = migraphx::argument(iou_thresh_shape, iou_thresh.data());
+
+    p.eval(pp).back();
+
+    EXPECT(true);
+}
+
```

Comment:
We don't normally parse an onnx file in ref_ops_test.cpp. You can copy and paste changing the `max_out_l` from TEST_CASE(nms_test) to test your change. I don't think the sanity test is necessary since nms_test should be fine. I'm pretty sure verify_onnx is only used for ONNX operators that we essentially map to other MIGX operators rather than have it's own separate MIGX op. For example, the BatchNormalization or Celu ONNX operators.